cfgov/v1/templatetags/mega_menu.py

```diff
@@ -0,0 +1,8 @@
+from v1.models.snippets import MenuItem
+from flags.template_functions import flag_enabled
+
+
+def get_menu_items(request):
+    show_draft = flag_enabled('DRAFT_MENU', request)
```

Comment:
If the goal is to automatically have draft changes show on draft previews (e.g. pages viewed on content), then it should be possible to leverage wagtail-sharing to do this. Rather than explicitly look for a sharing site, it'd be a bit cleaner to use the new [`before_serve_shared_page` hook](https://github.com/cfpb/wagtail-sharing#before_serve_shared_page) to add an attribute to the request, something like:

```py
@hooks.register('before_serve_shared_page')
def before_serve_shared_page(page, request, args, kwargs):
    request.show_draft_megamenu = True

def get_menu_items(request):
    if request.show_draft_megamenu:
        ...
```

(This would require an update of our wagtail-sharing integration to the latest release.)

I'm not sure whether this is better than using a feature flag, but it's another way to go.

Reply:
Ah, I wasn't aware of that hook! I think I'd prefer if it integrated with wagtail-sharing than use a feature flag. I don't know if this should be a blocker, but it should be something we consider changing in the future.

Reply:
I tried the `before_serve_shared_page` hook locally and this does seem like a cleaner approach than having another menu flag. The only thing I'm wondering about is if there are implications with this approach for using draft state on other servers like build? (Although I don't actually know if draft state would be used outside of previewing changes on content before they go to production...)

Reply:
Using that hook would prevent the use of the draft state on any server that doesn't have a sharing site set up. The build server does have one of these (look in admin under Settings, Sharing Sites). I'm okay with a tight link between sharing site and draft mega menu, as that feels consistent with how sharing sites are used.

Reply:
@chosak I think that makes sense. I added the draft state/sharing site association in https://github.com/cfpb/cfgov-refresh/pull/3396/commits/98c19fd5d9948c4e471e6c1a61e1ba37e32bb42c.